### ISSUE TITLE
[Bug] 2.4.0 Errors with 12 word Mnemonics, Regression in 2.4.0 as no errors in 2.3.1

### ISSUE THREAD

**Issue by BrandtH22** [Chia-Network/chia-blockchain#18243]
### What happened?

Initially reported by Nektodron in discord: https://discord.com/channels/1034523881404370984/1034870571864948777/1255165575597129788

Current workaround: downgrade to 2.3.1

Confirmed this is a regression in 2.4.0 as the below process works without error in previous versions.

Error: "Cannot read properties of undefined (reading 'pk')"

Steps to reproduce:
1. import 12 word mnemonics
2. Click logout
3. Select the 12 word mnemonic keyset and click Details
The imaged error will appear when the above is performed in 2.4.0 but in previous versions it properly displays the key information

![image](https://github.com/Chia-Network/chia-blockchain/assets/48610606/7ddaa843-e71e-4689-bf3b-a1749550509f)

### Version

2.4.0

### What platform are you using?

Windows

### What ui mode are you using?

GUI

### Relevant log output

```
2024-06-25T09:04:31.446 daemon chia.daemon.keychain_server: ERROR    
Traceback (most recent call last):
  File "chia\daemon\keychain_server.py", line 193, in handle_command
  File "chia\daemon\keychain_server.py", line 364, in get_key_for_fingerprint
  File "chia\util\keychain.py", line 274, in entropy
chia.util.errors.KeychainSecretsMissing
```

**Comment by BrandtH22** [comment]
Error in GUI seems to be due to the keyring not properly identifying the farmer or pool public keys:

```
Chia version 2.4.0

Label: test 12
Fingerprint: 688295223
Master public key (m): a9e652cb551d5978a9ee4b7aa52a4e826078a54b08a3d903c38611cb8a804a9a29c926e4f8549314a079e04ecde10cc1
Farmer public key (m/12381/8444/0/0): N/A
Pool public key (m/12381/8444/1/0): N/A
First wallet address: xch152m0ng6qwft4yfz67rmpgy9vct20p7v056te54vetc3ty5zcqehs9pfacf
```

When that same keyset is loaded in 2.3.1 they are present (and there is no error in the GUI):

```
Chia version: 2.3.1

Label: test 12
Fingerprint: 688295223
Master public key (m): a9e652cb551d5978a9ee4b7aa52a4e826078a54b08a3d903c38611cb8a804a9a29c926e4f8549314a079e04ecde10cc1
Farmer public key (m/12381/8444/0/0): 9849cb8aaa64a4b36a0fd3250e8886cbd92bec28b516f36b0c1571ad5617de20102926b53661300558196eb67e067e9d
Pool public key (m/12381/8444/1/0): 82846d7d05d60bad27fcf67394b92dab5b1d87fcdd31dc9cfb32d1273ebfdb981161330148451de3cc485f36f513cd98
First wallet address: xch152m0ng6qwft4yfz67rmpgy9vct20p7v056te54vetc3ty5zcqehs9pfacf
```

Note that a testing keyset was used for this so no concerns on keys being leaked.

Steps to reproduce the above:
NOTE- I did not re-import keys when switching versions.
- install 2.4.0
- import a 12 word mnemonic
- 2.4.0 has errors in the gui and cli does not show farmer or pool public keys
- uninstall 2.4.0
- install 2.3.1 (note keys are left entirely alone in this process)
- gui works as intended and pool plus farmer keys show in cli

**Comment by BrandtH22** [comment]
Can confirm that https://github.com/Chia-Network/chia-blockchain/pull/18246 fixes this issue :+1:

**Comment by wjblanke** [comment]
Fixed in 2.4.2rc1
